ignition: add butane link (#236)
flatcar-linux/flatcar-docs: docs/provisioning/ignition/specification.md

weight: 30
---

> Ignition uses a JSON format that is specified in several versions: v1, v2 and v3. While v1 and v2 are still supported in Flatcar Container Linux, it's recommended to write new configuration with v3.

Author: so this is something I was thinking about actually: documentation does not seem to follow the Flatcar's release cycle - served documentation is always the reflect of the current stable release.
User who's using alpha / beta / LTS releases has no way to use the documentation associated to a given release...
We could think about having the following links:
- https://www.flatcar.org/docs/latest/: stable
- https://www.flatcar.org/docs/alpha/: alpha
- https://www.flatcar.org/docs/beta/: beta
...

> Ignition uses a JSON format that is specified in several versions: v1, v2 and v3. While v1 and v2 are still supported in Flatcar Container Linux, it's recommended to write new configuration with v3.

Reviewer: The more meaningful way would be by major version, not by channel, I would say.
Due to the overhead this wasn't done.

I think it's good to state when a feature is available.
```suggestion
Ignition uses a JSON format that is specified in several versions: v1, v2 and v3. While v1 and v2 are still supported in Flatcar Container Linux, from version 3185.0.0, it's recommended to write new configuration with v3.
```

## Ignition v3

Starting from release 3185.0.0, Ignition v3 (specification 3.3.0) is supported in addition of Ignition v2. There are some things to be aware of:
* v1 and v2 are still supported and get translated at runtime; while this is tested well there may be corner cases where the v2 config relied on unspecified behavior
* `clevis` is not supported
* `kernelArguments` are supported and will persist the changes in `/usr/share/oem/grub.cfg` before the reboot but it only works for unconditional `set linux_append` statements in grub.cfg and `linux_console` is not considered

> * [butane][butane-spec] can be used to generate Ignition v3 specification

Reviewer: ```suggestion
* [butane][butane-spec] can be used to generate Ignition v3 configs
```

weight: 30
---

> Ignition uses a JSON format that is specified in several versions: v1, v2 and v3. While v1 and v2 are still supported in Flatcar Container Linux, from version 3185.0.0, it's recommended to write new configuration with v3.

Reviewer: ```suggestion
Ignition uses a JSON format that is specified in several versions: v1, v2 and v3 (which itself have minor versions like `2.3.0`). While v1 and v2 are still supported in Flatcar Container Linux, from version 3185.0.0, it's recommended to write new configuration with v3.
```

> Ignition uses a JSON format that is specified in several major versions: v1, v2 and v3 (which itself has minor versions like `2.3.0`). While v1 and v2 are still supported in Flatcar Container Linux, from version 3185.0.0, it's recommended to write new configuration with v3.

Reviewer 2: `v3` has minor versions like `2.3.0`? Or is it `v2`?

Reviewer: v2 and v3 just refer to the major version, like 2.3.0 and 3.3.0

## Ignition v2

> Ignition v2 is not developed anymore but still supported (specification 2.3.0), [ct][ct-config] can be used to emit Ignition v2 spec:

Reviewer: ```suggestion
Ignition v2 is not developed anymore but still supported (specification 2.3.0), the high-level [Container Linux Config YAML format][ct-config] can be used to emit Ignition v2 configs:
```
